test/e2e/nodemaintenance_quorumsize_test.go

```diff
@@ -0,0 +1,211 @@
+		nm := &operator.NodeMaintenance{}
+		err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: "nodemaintenance-xyz"}, nm)
+		if err != nil {
+			t.Logf("Failed to get %q nodeMaintenance: %v", nm.Name, err)
+			return false, nil
+		}
+
+		//t.Logf("%s: operator running state: %s", time.Now().Format("2006-01-02 15:04:05.000000"), nm.Status.Phase)
+		if nm.Status.Phase != operator.MaintenanceRunning {
+			t.Logf("%s: Running", time.Now().Format("2006-01-02 15:04:05.000000"))
+			return false, nil
+		}
+
+		if nm.Status.LastError != nmo.QuorumViolationErrorMsg {
+			t.Logf("%s: Running, with wrong error: %s", time.Now().Format("2006-01-02 15:04:05.000000"), nm.Status.LastError)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		showDeploymentStatus(t, f, fmt.Errorf("%s: Failed to verify running phase: %v", time.Now().Format("2006-01-02 15:04:05.000000"), err))
+	}
+
+	t.Logf("correct error has been observed")
+	t.Logf("delete nmo for node  %s", nodeName)
+
+	nodeMaintenanceDelete := &operator.NodeMaintenance{}
+
+	err = f.Client.Get(goctx.TODO(), types.NamespacedName{Namespace: nodeMaintenance.Namespace, Name: nodeMaintenance.Name}, nodeMaintenanceDelete)
+	if err != nil {
+		showDeploymentStatus(t, f, fmt.Errorf("Failed to get CRD. error %v", err))
+	}
+
+	// Delete the node maintenance custom resource
+	err = f.Client.Delete(goctx.TODO(), nodeMaintenanceDelete)
+	if err != nil {
+		showDeploymentStatus(t, f, fmt.Errorf("Could not delete node maintenance CR : %v", err))
+	}
+
+	if err := wait.PollImmediate(5*time.Second, 120*time.Second, func() (bool, error) {
+		nm := &operator.NodeMaintenance{}
+		err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: "nodemaintenance-xyz"}, nm)
+		if err == nil {
+			return false, nil
+		}
+		if !errors.IsNotFound(err) {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		showDeploymentStatus(t, f, fmt.Errorf("%s: Failed to verify that nmo has been deleted %v", time.Now().Format("2006-01-02 15:04:05.000000"), err))
+	}
+
+
+	node = &corev1.Node{}
+	err = f.Client.Get(goctx.TODO(), types.NamespacedName{Namespace: namespace, Name: nodeName}, node)
+	if err != nil {
+		showDeploymentStatus(t, f, fmt.Errorf("can't get node. error %v", err))
+	}
+
+	// Check that the deployment has 1 replica running after maintenance is removed.
+	t.Logf("%s: wait for deployment.", time.Now().Format("2006-01-02 15:04:05.000000"))
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, testDeployment, 1, retryInterval, timeout)
+	if err != nil {
+		showDeploymentStatus(t, f, fmt.Errorf("%s: failed to wait for deployment. error %v.", err, time.Now().Format("2006-01-02 15:04:05.000000")))
+	}
+
+	err = deleteSimpleDeployment(t, f, ctx, namespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("test deployment deleted")
+	t.Logf("test checkQuorumSizeViolation completed successfully")
+
+	err = f.Client.Delete(goctx.TODO(), pdb)
```

Comment:
Assuming we created it.  Which again, should not be the case on OCP 4

Reply:
we create it in all the test environments (including ocp 4.3) If that weren't the case then  createEtcPDB would have failed at line #59 with error.IsAlreadyExist.

Reply:
CI isn't the only place these tests can/should/will run, and kubevirtci isn't the only CI that could be used.
Marc already has a todo item for getting access to the OCP CI

Reply:
Makes sense. Added 1f312fed09860b330d20443707b378df43b68a1c that deletes the objects if they were created for the purpose of this test.